index.js

```diff
@@ -269,10 +269,26 @@ var PgDriver = Base.extend({
         var constraint = [],
             cb;
 
+        if ((spec.unique || spec.primaryKey) && spec.autoIncrement) {
+          switch (spec.dataType) {
+                case type.BIGINT:
+                    constraint.push('BIGSERIAL');                    
+                    break;
+                case type.SMALLINT:
+                    constraint.push('SMALLSERIAL');
+                    break;
+                default:
+                    constraint.push('SERIAL');
+                    break;
+            }          
+        }
+        else {
+          if (spec.autoIncrement) {
+            throw 'Auto increment column must be unique or primary key';
```

Comment:
Throwing here is not a good idea. I would rather go for rewriting those functions to return promises, to keep the error handling the default error handling.